pkg/cloudprovider/vsphereparavirtual/nsxipmanager/nsx_vpc.go

```diff
@@ -0,0 +1,96 @@
+package nsxipmanager
+
+import (
+	"context"
+
+	nsxapisv1alpha1 "github.com/vmware-tanzu/nsx-operator/pkg/apis/nsx.vmware.com/v1alpha1"
+	nsxclients "github.com/vmware-tanzu/nsx-operator/pkg/client/clientset/versioned"
+	nsxinformers "github.com/vmware-tanzu/nsx-operator/pkg/client/informers/externalversions"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+)
+
+const allocationSize = 24
+
+var _ NSXIPManager = &NSXVPCIPManager{}
+
+// NSXVPCIPManager is an implementation of NSXIPManager for NSX-T VPC.
+type NSXVPCIPManager struct {
+	client          nsxclients.Interface
+	informerFactory nsxinformers.SharedInformerFactory
+	svNamespace     string
+	ownerRef        *metav1.OwnerReference
+	podIPPoolType   string
+}
+
+// ClaimPodCIDR will claim pod cidr for the node by creating IPAddressAllocation CR.
+func (m *NSXVPCIPManager) ClaimPodCIDR(node *corev1.Node) error {
+	if _, err := m.informerFactory.Nsx().V1alpha1().IPAddressAllocations().Lister().IPAddressAllocations(m.svNamespace).Get(node.Name); err != nil {
```

Comment:
We can abstract the create function and refactor the structure to make it easier to read:
```go

if len(node.Spec.PodCIDRs) != 0 {
    klog.V(4).Infof("Pod CIDR %s is already set to node %s", node.Spec.PodCIDR, node.Name)
    return nil
}

if _, err := m.informerFactory.Nsx().V1alpha1().IPAddressAllocations().Lister().IPAddressAllocations(m.svNamespace).Get(node.Name); err != nil {
   if apierrors.IsNotFound(err) {
         return m.createIPAddressAllocation();
   } else {
         return err
   }
}

klog.V(4).Infof("Node %s already requested IPAddressAllocations", node.Name)
return nil

Reply:
Yes, it looks better. I copied code from ippool.